library/core/src/str/mod.rs

```diff
@@ -111,14 +111,7 @@ fn slice_error_fail_rt(s: &str, begin: usize, end: usize) -> ! {
     }
 
     // 2. begin <= end
-    assert!(
-        begin <= end,
-        "begin <= end ({} <= {}) when slicing `{}`{}",
-        begin,
-        end,
-        s_trunc,
-        ellipsis
-    );
+    assert!(begin <= end, "begin <= end ({begin} <= {end}) when slicing `{s_trunc}`{ellipsis}");
```

Comment:
This is much harder to parse, for me.

Reply:
Hm interesting, I find this much easier to read in one go, instead of having to jump back and forth between the `{}`s and the actual fmt arguments to figure out which one is the nth so it corresponds to the nth arg, etc.

Reply:
It's just the string `"(begin <= end, "begin <= end ({begin} <= {end})"` being here is confusing for me, it might actually be better if the identifier name was less repetitive with the messaging.

Reply:
I don't have any objection to "\`{s_trunc}\`{ellipsis}" for instance.

Reply:
`assert!((begin <= end), "begin ({begin}) <= end ({end}) when slicing..")` does that read better for you?

Reply:
...surprisingly yes.

Reply:
@workingjubilee since we had actual tests for this message, I wonder if it is actually better to leave it untouched altogether, what do you think?

Reply:
I'll sleep on it but my first reaction is that many tests in the test suite deliberately test the current behavior without trying to harden against future changes, on the thesis that testing the current behavior is preferable to writing a test that which would pass silently in the face of behavioral changes that might render it nonfunctional, and that in general it's fine to diff a test if you change the actual inputs to the test's criteria.